simulator.py

```diff
@@ -0,0 +1,97 @@
+"""
+Create a simulator to calculate pnl
+"""
+
+import copy
+import uuid
+from thousandaire.dataset import Data
+
+DATASET = 'price_dataset'
+
+def decode_data(data):
+    """
+    Decode while in multi-process.
+
+    In multi-process, simulator will get the data reference

```

Comment:
In multi-processing,

Reply:
the simulator will get data by shared memory, which are encoded by pickle to make them sharable.